src/Tizen.Diagnostics/Tizen.Diagnostics/Diagnostics.cs

```diff
@@ -0,0 +1,304 @@
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using NativeApi = Interop.Diagnostics;
+
+namespace Tizen.Diagnostics
+{
+    /// <summary>
+    /// The Diagnostics class is used for requesting client data and receiving diagnostics report events.
+    /// </summary>
+    /// <since_tizen> 8 </since_tizen>
+    public class Diagnostics : IDisposable
+    {
+        private bool _disposed = false;
+
+        /// <summary>
+        /// Event to be invoked when new report context arrives.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public event EventHandler<ReportReceivedEventArgs> ReportReceived;
+
+        /// <summary>
+        /// The Diagnostics class constructor.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public Diagnostics()
+        {
+            this.SetNotificationCallback();
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the Diagnostics class.
+        /// </summary>
+        ~Diagnostics()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Requests other apps to dump diagnostics data.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <param name="clientId">An id of app or service to request.</param>
+        /// <param name="parameters">Array of parameters passed to the client.</param>
+        /// <returns>
+        /// The instance of DiagnosticsData.
+        /// </returns>
+        /// <privilege>
+        /// http://tizen.org/privilege/internal/default/platform
+        /// </privilege>
+        public DiagnosticsData RequestClientData(string clientId, string[] parameters)
+        {
+            if (parameters is null)
+                throw new ArgumentNullException(nameof(parameters));
+
+            var ret = NativeApi.RequestClientData(clientId, parameters, parameters.Length, out IntPtr data);
+            if (ret != NativeApi.DiagnosticsError.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            return new DiagnosticsData(data);
+        }
+
+        private NativeApi.NotificationCallback _notificationCallback;
+
+        private void SetNotificationCallback()
+        {
+            _notificationCallback = OnNotification;
+            var ret = NativeApi.SetNotificationCb(_notificationCallback, IntPtr.Zero);
+            if (ret != NativeApi.DiagnosticsError.None)
+                throw ExceptionFactory.CreateException(ret);
+        }
+
+        private void OnNotification(IntPtr ctx, IntPtr data)
```

Comment:
Do you know lifetime of `IntPtr data`?
In Native API, Usually it only valid during on callback, but in C#, it should be valid out scope of callback

Reply:
It's valid until destroyed in Dispose function. Native API is not releasing this object, after returning from callback. It should be destroyed explicitly by (Native API) user.